src/perfcollect/perfcollect

```diff
@@ -1339,11 +1347,25 @@ SetupLTTngSession()
         elif [ "$events" == "threading" ]
         then
             EnableLTTngEvents ${DotNETRuntime_ThreadingKeyword[@]}
-        elif [ "$events" == "processlifetime" ]
-        then
-            # Setup kernel events. 
-            # TODO: enable triggers of different kernel events, instead of a default set of kernel events
-            EnableLTTngKernelEvents ${LTTng_Kernel_ProcessLifetime[@]}
+        else     
+            # Enable other keywords
+            events=(${events//","/" "})
+            for (( i=0; i<${#events[@]}; i++ ))
+            do 
+                local event=${events[$i]}
+                if [ "${!event}" == "" ]
+                then
+                    echo Invalid keyword $event, skipped...
+                    continue
+                fi
+                local -a 'keywords=("${'"$event"'[@]}")'
+                if [ "${event#"LTTng_Kernel_"}" != "$event" ]
```

Comment:
Seems completely reasonable.  One thing that would be good here is to make sure that the array exists as well so that EnableLTTngEvents doesn't blow up, or worse, just silently fail.

Reply:
That's right, will check if the array is empty instead of checking if the array variable exists.